zallet/src/components/json_rpc/methods/z_get_total_balance.rs

```diff
@@ -0,0 +1,76 @@
+use documented::Documented;
+use jsonrpsee::core::RpcResult;
+use schemars::JsonSchema;
+use serde::Serialize;
+use zcash_client_backend::data_api::WalletRead;
+
+use crate::components::{
+    database::DbConnection,
+    json_rpc::{server::LegacyCode, utils::value_from_zatoshis},
+};
+
+/// Response to a `z_gettotalbalance` RPC request.
+pub(crate) type Response = RpcResult<ResultType>;
+pub(crate) type ResultType = TotalBalance;
+
+/// The total value of funds stored in the wallet.
+#[derive(Clone, Debug, Serialize, Documented, JsonSchema)]
+pub(crate) struct TotalBalance {
+    /// The total value of unspent transparent outputs, in ZEC
+    transparent: f64,
+
+    /// The total value of unspent Sapling and Orchard outputs, in ZEC
+    private: f64,
+
+    /// The total value of unspent shielded and transparent outputs, in ZEC
+    total: f64,
```

Comment:
Bug: I just realised that these outputs are formatted explicitly as strings in `zcashd`, not JSON numbers as done here. I'll fix that at the same time as #15.